security/pkg/nodeagent/sds/server.go

```diff
@@ -101,6 +101,14 @@ func (s *Server) initDiscoveryService(options *Options, st SecretManager) error
 		return err
 	}
 
+	if _, err := os.Stat(options.UDSPath); err == nil {
+		log.Debugf("%q exists", options.UDSPath)
+
+		if err := os.Chmod(options.UDSPath, 0777); err != nil {
```

Comment:
Do we need the 'x' permission here? I feel 777 is an overkill.

Reply:
updated to 0666(jimmy helped to verified)